### PR TITLE
Dialogue: replace wp_kses with wp_kses_post

### LINKED ISSUE #18298: Dialog block: use `wp_kses_post`
Update `wp_kses` to `wp_kses_post` (https://github.com/Automattic/jetpack/pull/17916/files/99dc3b725145f948352d579b331a9d88ddbf0bb1#r545119072)

### PR DESCRIPTION
**Pull request by retrofox**
replace `wp_kses` with `wp_kses_post`.

Fixes #18298

#### Changes proposed in this Pull Request:

* Dialogue: replace wp_kses with wp_kses_post.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Check the dialogue content is parsed properly by the server-side, confirming that the block content looks good ni the front-end.

#### Proposed changelog entry for your changes:

n/a

### CODE REVIEW

**Comment by retrofox** [comment]
feedback https://github.com/Automattic/jetpack/pull/17916/files/99dc3b725145f948352d579b331a9d88ddbf0bb1#r545119072

File: projects/plugins/jetpack/extensions/blocks/dialogue/dialogue.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ function filter_content( $content ) {
 		return '';
 	}
 
-	return wp_kses(
+	return wp_kses_post(
 		$content,
 		array(
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/dialogue/dialogue.php:53]: You should be able to get rid of the second argument now, you don't need it with `wp_kses_post`.

Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/dialogue/dialogue.php:53]: You should be able to get rid of the second argument now, you don't need it with `wp_kses_post`.

**Review by jeherve — CHANGES_REQUESTED** [review]

**Review by jeherve — APPROVED** [review]

**Comment by retrofox** [comment]
r219647-wpcom

**Pull request merged**